backend/app/adapter/sqldb/userrole.go

```diff
@@ -30,7 +30,7 @@ WHERE "%s"=$1;
 		table.UserRole.ColumnUserID,
 	)
 
-	var roles []role.Role
+	roles := []role.Role{}
```

Comment:
Why do you prefer this more than var roles []role.Role?

Reply:
var initializes roles as a nil slice, this gives problem in assertion

`[]role.Role{} != []role.Role{nil}`